pkg/client/client.go

```diff
@@ -352,38 +352,42 @@ func (c *Client) DeleteConfigMap(cm *v1.ConfigMap) error {
 	return err
 }
 
-func (c *Client) DeleteHashedConfigMap(newHash, prefix string) error {
+// DeleteHashedConfigMap deletes all configmaps in the given namespace which have
+// the specified prefix, and DO NOT have the given hash.
+func (c *Client) DeleteHashedConfigMap(namespace, prefix, newHash string) error {
 	ls := "monitoring.openshift.io/name=" + prefix + ",monitoring.openshift.io/hash!=" + newHash
-	configMaps, err := c.KubernetesInterface().CoreV1().ConfigMaps(c.namespace).List(metav1.ListOptions{
+	configMaps, err := c.KubernetesInterface().CoreV1().ConfigMaps(namespace).List(metav1.ListOptions{
 		LabelSelector: ls,
 	})
 	if err != nil {
 		return errors.Wrapf(err, "error listing configmaps with label selector %s", ls)
 	}
 
-	for i := range configMaps.Items {
-		err := c.KubernetesInterface().CoreV1().ConfigMaps(c.namespace).Delete(configMaps.Items[i].Name, &metav1.DeleteOptions{})
+	for _, cm := range configMaps.Items {
+		err := c.KubernetesInterface().CoreV1().ConfigMaps(namespace).Delete(cm.Name, &metav1.DeleteOptions{})
 		if err != nil {
-			return errors.Wrapf(err, "error deleting configmap: %s", configMaps.Items[i].Name)
+			return errors.Wrapf(err, "error deleting configmap: %s", cm.Name)
```

Comment:
it might be good to include the namespace (same for for the return statement at L363).

Reply:
done